.changeset/thirty-birds-build.md

```diff
@@ -0,0 +1,32 @@
+---
+'@cloudflare/next-on-pages': patch
+---
+
+fix: implement route specific global scoping strategy
+
+currently routes all share the same global scope, this can be problematic and cause
+race conditions and failures
+
+One example of this is the following code that is present in route function files:
+
+```ts
+self.webpackChunk_N_E = ...
+```
+
+and
+
+```ts
+self.webpackChunk_N_E.push(...)
+```
+
+this indicates that an in-memory global collection of the webpack chunks is shared by all routes,
+this combined with the fact that chunks can have their own module state this can easily cause routes
+to conflict with each other at runtime.
+
+So, in order to solve the above issue wrap every route function in a function wrapper which
```

Comment:
Looks like some words are missing in this sentence

> So, in order to solve the above issue wrap every route function in a function wrapper which

Suggest

>  So, in order to solve the above issue, all route functions are wrapped in a function which

Reply:
well, words are not missing, maybe the sentence is just a bit unclear to follow 😅 

I tend to use the imperative tense for both the commit title and description (as I believe that to be a best practice on writing good commits), so sometimes the result can sound a bit forced 😕😓 

I'll try to improve the wording, while still keep it imperative 🙂 (unless you prefer descriptions not to be in imperative)

Reply:
I couldn't think of an imperative wording that would be much more clear than the current version 😕 

So I'm fine with what you suggested 👍